test/Microsoft.IdentityModel.Tokens.Tests/Validation/ValidatorsTests.cs

```diff
@@ -28,91 +28,89 @@ public void ValidateAudienceParameters(AudienceValidationTheoryData theoryData)
                     {
                         Audiences = new List<string> { "audience1" },
                         ExpectedException = ExpectedException.SecurityTokenInvalidAudienceException("IDX10208:"),
-                        TestId = "ValidAudienceEmptyString",
                         TokenValidationParameters = new TokenValidationParameters{ ValidAudience = "" }
                     },
-                    new AudienceValidationTheoryData
+                    new AudienceValidationTheoryData("ValidAudienceWhiteSpace")
                     {
                         Audiences = new List<string> { "audience1" },
                         ExpectedException = ExpectedException.SecurityTokenInvalidAudienceException("IDX10208:"),
-                        TestId = "ValidAudienceWhiteSpace",
                         TokenValidationParameters = new TokenValidationParameters{ ValidAudience = "    " }
                     },
-                    new AudienceValidationTheoryData
+                    new AudienceValidationTheoryData("ValidAudiencesEmptyString")
                     {
                         Audiences = new List<string> { "audience1" },
                         ExpectedException = ExpectedException.SecurityTokenInvalidAudienceException("IDX10214:"),
-                        TestId = "ValidAudiencesEmptyString",
                         TokenValidationParameters = new TokenValidationParameters{ ValidAudiences = new List<string>{ "" } }
                     },
-                    new AudienceValidationTheoryData
+                    new AudienceValidationTheoryData("ValidAudiencesWhiteSpace")
                     {
                         Audiences = new List<string> { "audience1" },
                         ExpectedException = ExpectedException.SecurityTokenInvalidAudienceException("IDX10214:"),
-                        TestId = "ValidAudiencesWhiteSpace",
                         TokenValidationParameters = new TokenValidationParameters{ ValidAudiences = new List<string>{ "    " } }
                     },
-                    new AudienceValidationTheoryData
+                    new AudienceValidationTheoryData("ValidateAudienceTrueValidAudienceAndValidAudiencesNull")
                     {
                         Audiences = new List<string> { "audience1" },
                         ExpectedException = ExpectedException.SecurityTokenInvalidAudienceException("IDX10208:"),
-                        TestId = "ValidateAudienceTrueValidAudienceAndValidAudiencesNull"
+                    },
+                    new AudienceValidationTheoryData("AudiencesEmpty_RequireAudienceFalse_NoException")
+                    // default value of TVP.RequireAudience is true.
+                    {
+                        Audiences = new List<string> { },
+                        TokenValidationParameters = new TokenValidationParameters{
+                            ValidAudience = "audience",
```

Comment:
Would we want the case also where there is no audience?

Reply:
In this test case, there is no audience (on line 109, Audiences is empty). It does hit the 10277 warning, but since there is no S2SContext to capture logs, it's not checked in this test case. When debugging, we do hit the warning and return out of the function (otherwise, a different error will be thrown since Audiences is empty). This affirms the expectation that we don't fault if an Audience isn't provided and TVP.RequireAudience is `false`.

Reply:
I added additional cases and made the original one more clear. I hope that helps, please reopen this comment needed.